Bangkok Post.js

```diff
@@ -0,0 +1,175 @@
+{
+	"translatorID": "7f74d823-d2ba-481c-b717-8b12c90ed874",
+	"label": "Bangkok Post",
+	"creator": "Matt Mayer",
+	"target": "https://www.bangkokpost.com/.*",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2020-09-07 19:12:01"
+}
+
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2020 Matt Mayer
+	
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	return 'newspaperArticle';
+}
+function doWeb(doc, url) {
+	scrape(doc, url);
+}
+function getMetaTag(doc,attr, value, contentattr) {
+	const tag = Array.from(doc.getElementsByTagName("meta")).filter(m => m.attributes[attr] && m.attributes[attr].value==value)[0]; 
+	if (tag && tag.attributes[contentattr]) {
+		return tag.attributes[contentattr].value;
+	}
+}
+function scrape(doc, url) {
+	const translator = Zotero.loadTranslator('web');
+	// Embedded Metadata
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+	translator.setDocument(doc);
+	
+	translator.setHandler('itemDone', function (obj, item) {
+		// Add data for fields that are not covered by Embedded Metadata
+		//author name is stored as firstname lastname
+		const authorName=getMetaTag(doc,"property","cXenseParse:author","content")
+		//date is stored as a timestamp like 2020-09-07T17:37:00+07:00, just extract the YYYY-MM-DD at start
+		if (authorName) {
+			item.creators= [ZU.cleanAuthor(authorName, "author",false)];
+		}
+
+		const date = getMetaTag(doc,"name","cXenseParse:recs:publishtime","content")
+		if (date) {
+			item.date = date.substr(0,10)
+		}
+		
+		item.publicationTitle = "Bangkok Post"
+		
+		
+		item.complete();
+	});
+
+	translator.getTranslatorObject(function(trans) {
+		trans.itemType = "newspaperArticle";
+		trans.doWeb(doc, url);
+	});
```

Comment:
I know there are many examples of using `getTranslatorObject()` like this in existing translators, but I'm not actually sure why — I haven't tested it, but I don't see why it should be necessary. As noted in the [documention](https://www.zotero.org/support/dev/translators/coding#using_gettranslatorobject), that's intended for situations where you need to access objects exposed by the child translator.

I think you should just be able to run `translator.translate();` and set the item type in the `itemDone` handler if necessary, as is done in the Washington Monthly translator:

https://github.com/zotero/translators/blob/c40859799b0091f4bbaecd43ffeb28834457ceee/Washington%20Monthly.js#L90-L119

Reply:
I think the reason you're seeing this pattern show up frequently it's because that's how it works in the tutorial on the Wikipedia page at https://m.mediawiki.org/wiki/Zotero_translator_for_Wikimedia_blog So probably a lot of people are copying from that

Reply:
simplified as suggested